### ISSUE TITLE
pb-rs no examples for how to own a structure after deserialize_from_slice

### ISSUE THREAD

**Issue by zutils** [tafia/quick-protobuf#134]
If I receive data in a function, and want to return a structure in my proto file, I must borrow from the &[u8]. See the following:

      fn my_func() -> FooBar {
          let bytes: Vec<u8> = ...;
          let foobar = deserialize_from_slice(&bytes).expect("Cannot convert into a `FooBar`");
          foobar
      }
 This function fails because "bytes" is being borrowed.

**Comment by nerdrew** [comment]
I have a branch that I've been using in a personal project that uses the rental crate (https://docs.rs/rental/0.5.3/rental/) to create protos that own the buffer they are parsed from. It actually feels pretty natural. I can open a PR if this approach sounds interesting.

It allows code like this that reads compressed proto bytes from a file and turns it into a proto:

```rust
fn open() -> Result<LogFileRental, Error> {
    let compressed = File::open("boom")?;
    let mut decoder = Decoder::new(compressed)?;
    let mut buf: Vec<u8> = Vec::new();
    decoder.read_to_end(&mut buf)?;
    trace!("read proto={:?}", &buf);
    Ok(buf.try_into()?)
}

fn main() {
    let proto = open().unwrap();
    // proto contains both the proto and the Vec<u8>.
    let proto_struct = proto.suffix();
    // use proto_struct as a normal proto struct.
}
```

**Comment by tafia** [comment]
Thanks @nerdrew .
I'm opened to have this under a feature flag if possible. I'd like to avoid it to be the default though.

**Comment by nerdrew** [comment]
Ya, in it's current form it is only a pb-rs flag. My app has the rental crate as a dependency already though, so I need to look at what that would look like.

**Comment by nerdrew** [comment]
Here's my branch with the rentals: https://github.com/tafia/quick-protobuf/pull/139. It isn't ready for a real review, but it shows the idea.

**Comment by nerdrew** [comment]
FWIW: https://github.com/tafia/quick-protobuf/pull/143 was merged which optionally generates "Owned" protos that should do what you want.

**Comment by tafia** [comment]
Just published v0.8.2!
